KA-Flocking/Assets/Scripts/FlockAgent.cs

```diff
@@ -9,32 +9,129 @@ public class FlockAgent : MonoBehaviour
     public float stability = 0.3f;
     // The speed at which the rotation of the agents stabilises
     public float stabilisationSpeed = 2.0f;
+    private readonly float acceleration = 15;
+    private readonly float maxAcceleration = 25;
     public Unit unit;
     Collider agentCollider;
-    Rigidbody rb;
+    public Rigidbody rb;
     public Collider AgentCollider { get { return agentCollider; } }
+    public Animator animator;
+    bool attacking = false;
+    Vector3 attackAlignment = Vector3.zero;
+    float attackCountDown = 0f;
+    public float dragCoefficient = 0.1f;
+    int animationMode = 0;
+    FightOrFlightBehaviour FOFH;
+    Vector3 accelerationBuffer = Vector3.zero;
+
 
     // Start is called before the first frame update
     void Start()
     {
         agentCollider = GetComponent<Collider>();
         rb = GetComponent<Rigidbody>();
+        animator = GetComponentInChildren<Animator>();
+        FOFH = ScriptableObject.CreateInstance<FightOrFlightBehaviour>();
     }
 
     public void Move(Vector3 acceleration)
```

Comment:
Please use some functional decomposition to make sure methods aren't 100 lines long.